core/core.js

```diff
@@ -0,0 +1,164 @@
+'use strict'; // For testing with node.js
+
+// Global scope declarations:
+var $ = Object.create(Object.prototype);
+
+(function() {
+  // Core build options:
+
+  // Overwrite existing?:
+  var force = false;
+
+  /**
+   * Add a property to a core object - but only overwrite existing
+   * properties if force is true.
+   * @param {Object} obj The object to which to add a property.
+   * @param {string} key The property key to add.
+   * @param {*} value The property key to add.
+   */
+  function make(obj, key, value) {
+    if (force || typeof obj[key] === 'undefined') {
+      obj[key] = value;
+    }
+  }
+
+  /*******************************************************************
+   * $.object
+   */
+  make($, 'object', Object.prototype);
+
+  /*******************************************************************
+   * $.physical
+   */
+  make($, 'physical', Object.create($.object, {
+    name: {value: 'Physical object prototype',
+	   writable: true, enumerable: true, configurable: true},
+    location: {value: null,
+	       writable: true, enumerable: true, configurable: true},
+    contents_: {value: undefined, // wil be vetted into existance
+		writable: true, enumerable: true, configurable: true},
+  }));
+
+  make($.physical, 'contents', function() {
+    // Return the contents of this object.  This is for VR purposes,
+    // and descendents of $.physical can override it to 'hide' certain
+    // objects from their contents.
+    $.physical_vet(this);
```

Comment:
I am thinking of this as a placeholder for $.physical.vet, awaiting non-overridable properties.